### PR TITLE
PYTHON-1366 About DocumentTooLarge - insert_one case

### PR DESCRIPTION
**Pull request by caseyclements**
This PR handles the first case of https://jira.mongodb.org/browse/PYTHON-1366 - insert_one(large_doc)
We no longer raise an Operation error if the document is larger than the document size limit but under the command size limit.

### CODE REVIEW

File: test/test_encryption.py [hunk 1]
```diff
@@ -1232,16 +1233,13 @@ def test_06_insert_fails_over_16MiB(self):
...
 
-        with self.assertRaisesRegex(WriteError, "object to insert too large"):
+        err_msg = "the connected server supports BSON document sizes up to 16777216 bytes."
+        with self.assertRaisesRegex(DocumentTooLarge, expected_regex=err_msg):
             self.coll_encrypted.insert_one(doc)
 
         # Same with bulk_write.
-        doc["_id"] = "encryption_exceeds_16mib_bulk"
-        with self.assertRaises(BulkWriteError) as ctx:
+        with self.assertRaisesRegex(DocumentTooLarge, expected_regex=err_msg):
```
Comment (ShaneHarvey) [inline test/test_encryption.py:1241]: Hmm this is surprising. Why does encrypted bulk_write raise DocumentTooLarge now but not regular bulk_write?

Comment (ShaneHarvey) [reply test/test_encryption.py:1241]: That doesn’t answer my question though.

**Comment by ShaneHarvey** [comment]
The link check failure is unrelated, the docs team is looking into it.

File: test/test_encryption.py (hunk 1 above)
Comment (caseyclements) [reply test/test_encryption.py:1241]: Where is the test for the regular bulk_write?

Comment (caseyclements) [reply test/test_encryption.py:1241]: Or somewhere in `test_bulk.py`?

**Pull request closed without merging**